tuf/tuf.go

```diff
@@ -965,62 +927,6 @@ func (tr *Repo) SignRoot(expires time.Time) (*data.Signed, error) {
 	return signed, nil
 }
 
-// get all the saved previous roles < the current root version
-func (tr *Repo) getOldRootRoles() versionedRootRoles {
-	oldRootRoles := make(versionedRootRoles, 0, len(tr.Root.Signed.Roles))
-
-	// now go through the old roles
-	for roleName := range tr.Root.Signed.Roles {
-		// ensure that the rolename matches our format and that the version is
-		// not too high
-		if data.ValidRole(roleName) {
-			continue
-		}
-		nameTokens := strings.Split(roleName, ".")
-		if len(nameTokens) != 2 || nameTokens[0] != data.CanonicalRootRole {
-			continue
-		}
-		version, err := strconv.Atoi(nameTokens[1])
-		if err != nil || version >= tr.Root.Signed.Version {
-			continue
-		}
-
-		// ignore invalid roles, which shouldn't happen
-		oldRole, err := tr.Root.BuildBaseRole(roleName)
-		if err != nil {
-			continue
-		}
-
-		oldRootRoles = append(oldRootRoles, versionedRootRole{BaseRole: oldRole, version: version})
-	}
-
-	return oldRootRoles
-}
-
-// gets any extra optional root keys from the existing root.json signatures
-// (because older repositories that have already done root rotation may not
-// necessarily have older root roles)
-func (tr *Repo) getOptionalRootKeys(signingRoles []data.BaseRole) []data.PublicKey {
-	oldKeysMap := make(map[string]data.PublicKey)
-	for _, oldSig := range tr.Root.Signatures {
-		if k, ok := tr.Root.Signed.Keys[oldSig.KeyID]; ok {
-			oldKeysMap[k.ID()] = k
-		}
-	}
-	for _, role := range signingRoles {
-		for keyID := range role.Keys {
-			delete(oldKeysMap, keyID)
-		}
-	}
-
-	oldKeys := make([]data.PublicKey, 0, len(oldKeysMap))
-	for _, key := range oldKeysMap {
-		oldKeys = append(oldKeys, key)
-	}
-
-	return oldKeys
-}
-
 func oldRootVersionName(version int) string {
```

Comment:
Followup PR: I think we can remove this function. I don't think it's used anywhere with the new root rotation logic.